### PR TITLE
Fix DefaultCodeExecTaskExecutorManager.instanceToExecutors concurrency issues

### PR DESCRIPTION
**Pull request by peacewong**
### What is the purpose of the change

1. optimize concurrentExecutor idle exit
2. Fix DefaultCodeExecTaskExecutorManager.instanceToExecutors concurrency issues

### Related issues/PRs

Related issues:  #4078,#4117

### Brief change log

1. optimize concurrentExecutor idle exit
2. Fix DefaultCodeExecTaskExecutorManager.instanceToExecutors concurrency issues

### Checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [Linkis mailing list](https://linkis.apache.org/community/how-to-subscribe) first)
- [ ] **If this is a code change**: I have written unit tests to fully verify the new behavior.

### CODE REVIEW

File: linkis-computation-governance/linkis-engineconn/linkis-computation-engineconn/src/main/scala/org/apache/linkis/engineconn/computation/executor/execute/ConcurrentComputationExecutor.scala [hunk 1]
```diff
@@ -26,11 +26,46 @@ abstract class ConcurrentComputationExecutor(override val outputPrintLimit: Int
...
+  override def execute(engineConnTask: EngineConnTask): ExecuteResponse = {
+    if (isBusy) {
+      logger.error(
+        s"Executor is busy but still got new task ! Running task num : ${getRunningTask}"
+      )
+    }
+    if (getRunningTask >= getConcurrentLimit) EXECUTOR_STATUS_LOCKER.synchronized {
+      if (getRunningTask >= getConcurrentLimit && NodeStatus.isIdle(getStatus)) {
+        logger.info(
+          s"running task($getRunningTask) > concurrent limit $getConcurrentLimit, now to mark engine to busy "
```
Comment (casionone) [inline linkis-computation-governance/linkis-engineconn/linkis-computation-engineconn/src/main/scala/org/apache/linkis/engineconn/computation/executor/execute/ConcurrentComputationExecutor.scala:40]: `running task: $getRunningTask > concurrent limit: $getConcurrentLimit, now to mark engine to busy`
or `running task($getRunningTask) > concurrent limit($getConcurrentLimit), now to mark engine to busy`?

File: linkis-orchestrator/linkis-computation-orchestrator/src/main/scala/org/apache/linkis/orchestrator/computation/conf/ComputationOrchestratorConf.scala [hunk 2]
```diff
@@ -40,10 +40,7 @@ object ComputationOrchestratorConf {
     "rm,sh,find,kill,python,for,source,hdfs,hadoop,spark-sql,spark-submit,pyspark,spark-shell,hive,yarn"
   )
 
-  val SHELL_WHITE_USAGE = CommonVars(
-    "wds.linkis.shell.white.usage",
-    "sqoop,cd,ll,ls,echo,cat,tree,diff,who,grep,whoami,set,pwd,cut,file,head,less,if,while"
-  )
+  val SHELL_WHITE_USAGE = CommonVars("wds.linkis.shell.white.usage", "cd,ls")
```
Comment (casionone) [inline linkis-orchestrator/linkis-computation-orchestrator/src/main/scala/org/apache/linkis/orchestrator/computation/conf/ComputationOrchestratorConf.scala:44]: need update docs/configuration/linkis-computation-orchestrator.md  of wds.linkis.shell.white.usage  info

File: linkis-orchestrator/linkis-computation-orchestrator/src/main/scala/org/apache/linkis/orchestrator/ecm/ComputationEngineConnManager.scala [hunk 3]
```diff
@@ -105,13 +105,13 @@ class ComputationEngineConnManager extends AbstractEngineConnManager with Loggin
         }
       } catch {
         case t: LinkisRetryException =>
-          val taken = ByteTimeUtils.msDurationToString(System.currentTimeMillis - start)
+          val taken = System.currentTimeMillis() - start
           logger.warn(
             s"${mark.getMarkId()} Failed to askEngineAskRequest time taken ($taken), ${t.getMessage}"
```
Comment (casionone) [inline linkis-orchestrator/linkis-computation-orchestrator/src/main/scala/org/apache/linkis/orchestrator/ecm/ComputationEngineConnManager.scala:110]: ` s"${mark.getMarkId()} Failed to askEngineAskRequest time taken ($taken), ${t.getMessage}"`
->
 s"${mark.getMarkId()} Failed to askEngineAskRequest time taken: $taken ms, ${t.getMessage}"

**Comment by peacewong** [comment]
Thanks，updated

**Review by casionone — APPROVED** [review]

**Pull request merged**